src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoBean.java

```diff
@@ -0,0 +1,134 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.system.status;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.geotools.util.logging.Logging;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.stereotype.Component;
+
+/**
+ * *
+ *
+ * <p>It's possible to extend this class to change or add other low level APIs and use this new
+ * class as main collector after register using autowiring.
+ *
+ * @author d.stueken
+ */
+@Component
+public class OSHISystemInfoBean extends BaseSystemInfoCollector implements DisposableBean {
```

Comment:
Please don't annotate as @Component, gs-main is not using autowiring and package scan (newer modules do, but when they do, the whole module is auto wired).

Also, please don't introduce the "Bean" naming convention, check the existing applicationContext.xml files to double-check that indeed we do not normally call Spring beans "xyzBean".

I'm not saying we should not do it, but if we did, it would have to be uniform across the project, we don't want each developer to go their merry way.